subsys/bluetooth/mesh/shell/brg_cfg.c

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/bluetooth/mesh.h>
+#include <zephyr/bluetooth/mesh/shell.h>
+
+#include "mesh/foundation.h"
+#include "utils.h"
+
+static int cmd_subnet_bridge_get(const struct shell *sh, size_t argc, char *argv[])
+{
+	enum bt_mesh_subnet_bridge_state rsp;
+	int err;
+
+	err = bt_mesh_brg_cfg_cli_subnet_bridge_get(bt_mesh_shell_target_ctx.net_idx,
+						    bt_mesh_shell_target_ctx.dst, &rsp);
+	if (err) {
+		shell_error(sh, "Failed to send Subnet Bridge Get (err %d)", err);
+		return 0;
```

Comment:
This is quite strange why 0 is returned in case of failed execution.
According to shell API:
```
/**
 * @brief Shell command handler prototype.
 *
 * @param sh Shell instance.
 * @param argc  Arguments count.
 * @param argv  Arguments.
 *
 * @retval 0 Successful command execution.
 * @retval 1 Help printed and command not executed.
 * @retval -EINVAL Argument validation failed.
 * @retval -ENOEXEC Command not executed.
 */
```
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/shell/shell.h#L162-L175

I see this was implemented in such manner in many other places in mesh shell commands. However, there are correct implementations.

Reply:
I agree that it is strange! Do you think returning `-ENOEXEC` is sufficient?

Reply:
I think yes. It will follow shell engine documentation.